tests/integration/test_reads.py

```diff
@@ -473,6 +474,31 @@ def test_sanitize_character(catalog: Catalog) -> None:
     assert arrow_table.schema.names[0] == table_test_table_sanitized_character.schema().fields[0].name
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize('catalog', [pytest.lazy_fixture('session_catalog_hive'), pytest.lazy_fixture('session_catalog')])
+def test_sanitize_character_partitioned(catalog: Catalog) -> None:
+    table_name = "default.test_table_partitioned_sanitized_character"
+    try:
+        catalog.drop_table(table_name)
+    except NoSuchTableError:
+        pass
+
+    tbl = catalog.create_table(
+        identifier=table_name, schema=Schema(NestedField(field_id=1, name="some.id", type=IntegerType(), required=True))
+    )
```

Comment:
Seems it works! How about we move https://github.com/apache/iceberg-python/blob/35d4648f38c1ae924dbf9597d707e44f7c53225a/tests/integration/test_writes/utils.py#L62-L86
to a place that can be shared with this test in `test_reads.py`?

Alternatively, we can move this to `test_writes.py` since this test equally covers write side too.

Reply:
Sorry for the late reply here @HonahX. Great suggestion, I've moved the test 👍